### ISSUE TITLE
[Bug]: WoW Patch 11.0.2.56011 error on load into game

### ISSUE THREAD

**Issue by fraerie** [Caedilla/RSA#26]
### Description

Multiple LUA errors on game load in

### RSA Version

RSA 5.2

### World of Warcraft Flavor

Dragonflight

### Lua Error

2x RSA/Options.lua:272: attempt to call global 'GetAddOnMetadata' (a nil value)
[string "@RSA/Options.lua"]:272: in function <RSA/Options.lua:268>
[string "@RSA/Options.lua"]:2283: in function `RegisterOptions'
[string "@RSA/Options.lua"]:2322: in function <RSA/Options.lua:2316>
[string "=[C]"]: ?
[string "@Ace3/AceAddon-3.0-13/AceAddon-3.0.lua"]:66: in function <Ace3/AceAddon-3.0/AceAddon-3.0.lua:61>
[string "@Ace3/AceAddon-3.0-13/AceAddon-3.0.lua"]:494: in function `InitializeAddon'
[string "@Ace3/AceAddon-3.0-13/AceAddon-3.0.lua"]:619: in function <Ace3/AceAddon-3.0/AceAddon-3.0.lua:611>

Locals:
(*temporary) = <table> {
 type = "group"
}
(*temporary) = "RSA [|c5500DBBDRaeli's Spell Announcer|r] |c5500DBBD"
(*temporary) = nil
(*temporary) = "RSA"
(*temporary) = "Version"
(*temporary) = "attempt to call global 'GetAddOnMetadata' (a nil value)"
L = <table> {
 How long before this fake event triggers after any other event for this spell has been processed. = "How long before this fake event triggers after any other event for this spell has been processed."
 When this buff or debuff is expires. = "When this buff or debuff is expires."
 General Replacement = "General Replacement"
 Prevents multiple RSA users from announcing this spell. = "Prevents multiple RSA users from announcing this spell."
 End = "End"
 CC Broken = "CC Broken"
 Damage = "Damage"
 Configure each spell's announcement settings, such as what channels to announce in and what messages to send. = "Configure each spell's announcement settings, such as what channels to announce in and what messages to send."
 Remove Spell = "Remove Spell"
 Spell Ends = "Spell Ends"
 Missing options. Please report this! = "Missing options. Please report this!"
 Enable in the non-instanced world area when playing with War Mode %s. = "Enable in the non-instanced world area when playing with War Mode %s."
 Enable in Group Finder Dungeons = "Enable in Group Finder Dungeons"
 PvE = "PvE"
 Summon = "Summon"
 Spell ID = "Spell ID"
 |cffFFCC00Whispers|r the target of the spell. = "|cffFFCC00Whispers|r the target of the spell."
 |cff91BE0F/party|r if you're in a manually formed group. = "|cff91BE0F/party|r if you're in a manually formed group."
 Configuring:|r %s = "Configuring:|r %s"
 Resurrect = "Resurrect"
 Enable in Combat = "Enable in Combat"
 %s only while grouped = "%s only while grouped"
 %s can only function inside instances since 8.2.5. = "%s can only function inside instances since 8.2.5."
 Dispel = "Dispel"
 Aura Applied = "Aura Applied"
 Enable in Group Finder Raids = "Enable in Group Finder Raids"
 Event unique spell ID = "Event unique spell ID"
 Cannot configure while in combat. = "Cannot configure while in combat."
 When this resurrection spell finishes, giving the target the option to return to life. = "When this resurrection spell finishes, giving the target the option to return to life."
  If you wish to add a message for this section, enter it above in the |cffFFD100Add New Message|r box. As no messages exist, nothing will be announced for this section. = " If you wish to add a message for this section, enter it above in the |cffFFD100Add New Message|r box. As no messages exist, nothing will be announced for this section."
 List of Additional Spell IDs = "List of Additional Spell IDs"
 Custom Caster = "Custom Caster"
 When this spell captures a buff from the target. = "When this spell captures a buff from the target."
 Enable in scenario instances. = "Enable in scenario instances."
 When the target is immune to your spell. = "When the target is immune to your spell."
 Always uses spell target's name = "Always uses spell target's name"
 If selected, |c5500DBBD[TARGET]|r will always use the spell target's name, rather than using the input below for whispers. = "If selected, |c5500DBBD[TARGET]|r will always use the spell target's name, rather than using the input below for whispers."
  RSA will choose a message from this section at random, if you wish to remove a message, delete the contents and press enter. If no messages exist, nothing will be announced for this section. = " RSA will choose a message from this section at random, if you wish to remove a message, delete the contents and press enter. If no messages exist, nothing will be announced for this section."
 Unlock setup = "Unlock setup"
 Channel Name = "Channel Name"
 Add Event = "Add Event"
 Spell Setup for this spell is locked. = "Spell Setup for this spell is locked."
 Spell Starts = "Spell Starts"
 If this event uses a different spell ID to the primary one, enter it here. = "If this event uses a different spell ID to the primary one, enter it here."
 When the target absorbs your spell |c5500DBBD[MISST

Also:
2x RSA/Init.lua:156: attempt to call global 'GetAddOnMetadata' (a nil value)
[string "@RSA/Init.lua"]:156: in function <RSA/Init.lua:133>
[string "=[C]"]: ?
[string "@Ace3/AceAddon-3.0-13/AceAddon-3.0.lua"]:66: in function <Ace3/AceAddon-3.0/AceAddon-3.0.lua:61>
[string "@Ace3/AceAddon-3.0-13/AceAddon-3.0.lua"]:494: in function `InitializeAddon'
[string "@Ace3/AceAddon-3.0-13/AceAddon-3.0.lua"]:619: in function <Ace3/AceAddon-3.0/AceAddon-3.0.lua:611>

Locals:
self = <table> {
 modules = <table> {
 }
 baseName = "RSA"
 configData = <table> {
 }
 defaultModuleState = true
 Monitor = <table> {
 }
 db = <table> {
 }
 enabledState = true
 defaultModuleLibraries = <table> {
 }
 name = "RSA"
 orderedModules = <table> {
 }
 Comm = <table> {
 }
 Options = <table> {
 }
 monitorData = <table> {
 }
 SendMessage = <table> {
 }
}
defaults = <table> {
 profile = <table> {
 }
}
(*temporary) = <table> {
 version = "5.2"
 personalID = "function: 0000018AA4869C70692"
}
(*temporary) = nil
(*temporary) = "RSA"
(*temporary) = "Version"
(*temporary) = "attempt to call global 'GetAddOnMetadata' (a nil value)"
RSA = <table> {
 modules = <table> {
 }
 baseName = "RSA"
 configData = <table> {
 }
 defaultModuleState = true
 Monitor = <table> {
 }
 db = <table> {
 }
 enabledState = true
 defaultModuleLibraries = <table> {
 }
 name = "RSA"
 orderedModules = <table> {
 }
 Comm = <table> {
 }
 Options = <table> {
 }
 monitorData = <table> {
 }
 SendMessage = <table> {
 }
}
uClass = "druid"

Also:
12x RSA/Tools/Helpers.lua:120: attempt to call global 'GetSpellInfo' (a nil value)
[string "@RSA/Tools/Helpers.lua"]:120: in function `GetSpellInfo'
[string "@RSA/SpellData/DeathKnight.lua"]:266: in main chunk

Locals:
id = 123981
(*temporary) = nil
(*temporary) = 123981
(*temporary) = "attempt to call global 'GetSpellInfo' (a nil value)"

### Reproduction Steps

Load into game

**Comment by Farmbuyer** [comment]
A bunch of these are the removal of the deprecated function references, e.g., [GetAddOnMetadata](https://warcraft.wiki.gg/wiki/API_GetAddOnMetadata).

Fixing those the load process a little further, but then it runs into 

```
2x RSA/SpellData/Shaman.lua:152: attempt to concatenate a table value
[string "@RSA/SpellData/Shaman.lua"]:152: in main chunk

Locals:
RSA = <table> {
 modules = <table> {
 }
 configData = <table> {
 }
 defaultModuleState = true
 enabledState = true
 defaultModuleLibraries = <table> {
 }
 name = "RSA"
 orderedModules = <table> {
 }
 Comm = <table> {
 }
 monitorData = <table> {
 }
 SendMessage = <table> {
 }
}
L = <table> {
```

and I haven't looked any further.

**Comment by raptormama** [comment]
GetAddOnMetadata is now C_AddOns.GetAddOnMetadata. Several functions changed namespaces in this patch.

**Comment by Caedilla** [comment]
Should be fixed in 5.3

**Comment by Caedilla** [comment]
Should be fixed in 5.3
